app/controllers/application_controller.rb

```diff
@@ -9,6 +9,11 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  # prepend_before_action is being used in devise not before_action
+  # we're attempting to have prepend_before_action happen for just devise powered pages
+  # while before_action happens for all other pages
+  prepend_before_action :set_locale
```

Comment:
I think you could do this whole thing as:

```ruby
prepend_before_action :set_locale, if: :devise_controller?
before_action :set_locale, unless: :devise_controller?
```
but do you know if anything else breaks if you just use `prepend_before_action` for everything?

Reply:
I attempted this but it was breaking for some reason. I'll take a look at the `devise-i18n` gem

Reply:
This shouldn't be working either, unfortunately, because `if :devise_controller?` will always return true (symbols are truthy, apparently), so this is only working by accident right now. That's why you need to use the `if:` and `unless:` symbols as arguments. We can pair if you want to figure out why things were breaking/what was breaking when you did this the other way.